cloud-azure/src/test/java/com/sequenceiq/cloudbreak/cloud/azure/AzureUtilsTest.java

```diff
@@ -27,12 +44,56 @@ public void shouldAdjustResourceNameLengthIfItsTooLong() {
         CloudContext context = new CloudContext(7899L, "thisisaverylongazureresourcenamewhichneedstobeshortened", "dummy1", "dummy2");
 
         //WHEN
-        String testResult = subject.getStackName(context);
+        String testResult = underTest.getStackName(context);
 
         //THEN
         Assert.assertNotNull("The generated name must not be null!", testResult);
         assertEquals("The resource name is not the excepted one!", "thisisaverylongazureresourcenamewhichneedstobe7899", testResult);
         Assert.assertTrue("The resource name length is wrong", testResult.length() == Integer.parseInt(MAX_RESOURCE_NAME_LENGTH));
 
     }
+
+    @Test
+    public void validateStorageTypeForGroupWhenPremiumStorageConfiguredAndFlavorNotPremiumThenShouldThrowCloudConnectorException() {
+        thrown.expect(CloudConnectorException.class);
+
+        String flavor = "Standard_A10";
+        AzureDiskType azureDiskType = AzureDiskType.PREMIUM_LOCALLY_REDUNDANT;
+
+        when(azurePremiumValidatorService.premiumDiskTypeConfigured(azureDiskType)).thenReturn(true);
+        when(azurePremiumValidatorService.validPremiumConfiguration(flavor)).thenReturn(false);
+
+        underTest.validateStorageTypeForGroup(azureDiskType, flavor);
+
+        verify(azurePremiumValidatorService, times(1)).premiumDiskTypeConfigured(azureDiskType);
+        verify(azurePremiumValidatorService, times(1)).validPremiumConfiguration(flavor);
+
+    }
+
+    @Test
+    public void validateStorageTypeForGroupWhenPremiumStorageNotConfiguredThenShouldNotCallInstanceValidation() {
+        String flavor = "Standard_A10";
+        AzureDiskType azureDiskType = AzureDiskType.GEO_REDUNDANT;
+
+        when(azurePremiumValidatorService.premiumDiskTypeConfigured(azureDiskType)).thenReturn(false);
+
+        underTest.validateStorageTypeForGroup(azureDiskType, flavor);
+
+        verify(azurePremiumValidatorService, times(1)).premiumDiskTypeConfigured(azureDiskType);
```

Comment:
I would add a verify for that the instance validation never happened:
verify(azurePremiumValidatorService, times(0)).validPremiumConfiguration(flavor);